src/components/flame-graph/Canvas.js

```diff
@@ -58,6 +58,38 @@ class FlameGraphCanvas extends React.PureComponent<Props> {
     (this: any)._hitTest = this._hitTest.bind(this);
   }
 
+  getColor(implementation: Implementation, selfTimeRelative: number): string {
+    let h, s, l;
+    // The more self time a function has, the darker the color
+    // returned.  Do this by subtracting a lightness value from a base
+    // color for a particular implementation.  `a` and `b` are factors
+    // used in calcuating how much ligthness to subtract.  `limit` is
+    // the lower limit of the lightness.
+    let a, b, limit;
+
+    if (implementation === 'cpp') {
+      [h, s, l] = [358, 98, 82];
+      a = 4;
+      b = 5;
+      limit = 55;
+    } else if (implementation === 'js') {
+      [h, s, l] = [188, 53, 71];
+      a = 8;
+      b = 6;
+      limit = 40;
+    } else {
+      // Unknown implementation
+      [h, s, l] = [77, 70, 71];
+      a = 10;
+      b = 10;
+      limit = 40;
+    }
```

Comment:
So these three base colors were chosen from paletton.com. I tried to choose three colors that would account for the most common forms of color blindness, but I have no idea whether I was successful there.

I have empirically found values for the factors `a` and `b`, and `limit` by visual inspection (and some initial curve fitting), and trying to subjectively determine what looks nice. I started with a linear formula, but using a logarithm appeared to me to give a better range of colors for the gradient.

Reply:
I think this observation would be great as a code comment.